projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,229 @@
+ */
+
+use Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Abstract_Token_Subscription_Service;
+use const Automattic\Jetpack\Extensions\Subscriptions\META_NAME_FOR_POST_LEVEL_ACCESS_SETTINGS;
+
+/**
+ * Jetpack_Subscribe_Overlay class.
+ */
+class Jetpack_Subscribe_Overlay {
+	/**
+	 * Jetpack_Subscribe_Overlay singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Overlay|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Overlay instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Overlay();
+		}
+
+		return self::$instance;
+	}
+
+	const BLOCK_TEMPLATE_PART_SLUG = 'jetpack-subscribe-overlay';
+
+	/**
+	 * Jetpack_Subscribe_Overlay class constructor.
+	 */
+	public function __construct() {
+		if ( apply_filters( 'jetpack_subscribe_overlay_enabled', false ) ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+			add_action( 'wp_footer', array( $this, 'add_subscribe_overlay_to_frontend' ) );
+		}
+
+		add_filter( 'get_block_template', array( $this, 'get_block_template_filter' ), 10, 3 );
+	}
+
+	/**
+	 * Returns the block template part ID.
+	 *
+	 * @return string
+	 */
+	public static function get_block_template_part_id() {
+		return get_stylesheet() . '//' . self::BLOCK_TEMPLATE_PART_SLUG;
+	}
+
+	/**
+	 * Makes get_block_template return the WP_Block_Template for the Subscribe Overlay.
+	 *
+	 * @param WP_Block_Template $block_template The block template to be returned.
+	 * @param string            $id Template unique identifier (example: theme_slug//template_slug).
+	 * @param string            $template_type Template type: `'wp_template'` or '`wp_template_part'`.
+	 *
+	 * @return WP_Block_Template|null
+	 */
+	public function get_block_template_filter( $block_template, $id, $template_type ) {
+		if ( empty( $block_template ) && $template_type === 'wp_template_part' ) {
+			if ( $id === self::get_block_template_part_id() ) {
+				return $this->get_template();
+			}
+		}
+
+		return $block_template;
+	}
+
+	/**
+	 * Returns a custom template for the Subscribe Overlay.
+	 *
+	 * @return WP_Block_Template
+	 */
+	public function get_template() {
+		$template                 = new WP_Block_Template();
+		$template->theme          = get_stylesheet();
+		$template->slug           = self::BLOCK_TEMPLATE_PART_SLUG;
+		$template->id             = self::get_block_template_part_id();
+		$template->area           = 'uncategorized';
+		$template->content        = $this->get_subscribe_overlay_template_content();
+		$template->source         = 'plugin';
+		$template->type           = 'wp_template_part';
+		$template->title          = __( 'Jetpack Subscribe overlay', 'jetpack' );
+		$template->status         = 'publish';
+		$template->has_theme_file = false;
+		$template->is_custom      = true;
+		$template->description    = __( 'An overlay that shows up when someone visits your site', 'jetpack' );
```

Comment:
Is the copy consistent with the toggle setting label from designs?

Reply:
Good question. The toggle label is:

<img width="1179" alt="Screenshot 2024-05-15 at 09 49 12" src="https://github.com/Automattic/jetpack/assets/4068554/d5b49323-5029-4c22-8195-003db0595d4d">

Which IMO is missing the verb, like "Enable" or sth. Now it's a bit mysterious to me, @crisbusquets WDYT?

Reply:
Although I see why it can be helpful, I wouldn't use `Enable` because then all the toggles will begin the same way, and it'll be repetitive.

Almost none of the Jetpack Settings toggles include a verb, even though all of them are enabling something.

<img width="918" alt="Screenshot 2024-05-16 at 10 52 58" src="https://github.com/Automattic/jetpack/assets/25607244/f11c30b8-d510-4a90-8f06-6baed404fc51">

Reply:
Sure, what about the template description? It's visible when a user is editing it.

<img width="928" alt="Screenshot 2024-05-16 at 11 01 58" src="https://github.com/Automattic/jetpack/assets/4068554/2a011bd7-624f-4660-8cb8-3b332c8d775f">

Reply:
The description needs `.` at the end.

Reply:
Added, the subscribe modal and modal for comments don't have the dot as well. Could add it there as well but it would break translations.

Reply:
Translations would continue work — translation team applies "fuzzy" label to strings that change only 1-2 characters. You can always confirm with them it's the case with this specific string too.

Reply:
Ah, didn't know that, thanks!